modules/programs/tealdeer.nix

```diff
@@ -4,29 +4,59 @@ with lib;
 let
   cfg = config.programs.tealdeer;
 
-  tomlFormat = pkgs.formats.toml { };
-
   configDir = if pkgs.stdenv.isDarwin then
     "Library/Application Support"
   else
     config.xdg.configHome;
 
+  tomlFormat = pkgs.formats.toml { };
+
+  settingsFormat = let
+    updatesSection = types.submodule {
+      options = {
+        auto_update = mkOption {
+          type = types.bool;
```

Comment:
I'm not sure about this. Should I make this `nullOr bool`?
Currently, this mimics the default [settings](https://dbrgn.github.io/tealdeer/config_updates.html#auto_update), but in the future, it might change. Maybe it's better to not produce a config file at all?

Reply:
Can't remember if this is possible, but maybe we can make the `options` submodule actually default to `null` and just not include it if none of the options are specified?

Reply:
@sumnerevans please have another look. I made the submodule of type `types.nullOr settingsFormat;` with the default value of `null`